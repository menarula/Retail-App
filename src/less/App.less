.my-retail {
	padding: 50px;
	&-product {
		text-align: center;
	}
	&-title {
		font-size: 26px;
		width: 59%;
		margin: auto;
		line-height: 1.2em;
	}
	&-product-image-wrapper {
    width: 200px;
    height: 200px;
		margin: auto;
		padding: 20px 0;
	}
	&-product-image {
		display: block;
    width: auto;
    height: auto;
    max-width: 100%;
    max-height: 100%
	}
	&-price {
		font-size: 30px;
		font-weight: bold;
		margin-bottom: 10px;
	}
	&-price-qualifier {
		padding-left: 5px;
		font-size: 13px;
		font-weight: normal;
		text-transform: lowercase;
	}
	&-image-carousel {
		width: 200px;
    height: 100px;
		margin: auto;
	}
	&-promotions {
		padding: 10px 0;
		width: 100%;
		color: #cc0000;
		border-top: 1px solid #d3d3d3;
		border-bottom: 1px solid #d3d3d3;
	}
	&-button-wrapper {
		margin-top: 20px;
		padding: 0;
		.btn-sm {
			padding: 6px;
			font-size: 13px;
		}
		.btn-pickup {
			float: left;
			margin-right: 10px;
		}
		.btn-block {
			margin-top: 0;
			width: 45%;
		}
	}
	.btn-add-to-cart {
		color: #fff;
		background-color: #c00;
		border: 1px #c00 solid;
	}
	.btn-pickup {
		color: #fff;
		background-color: #000000;
		border: 1px #000000 solid;
	}
}
.carousel .slide {
	min-width: 33%;
}
@media screen and (max-width: 768px) {
    .my-retail-title {
      font-size: 18px;
			width: 88%;
    }
		.my-retail-button-wrapper .btn-block {
			font-size: 11px;
		}
}
