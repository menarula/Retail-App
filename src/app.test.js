import React from 'react';
import Enzyme, { mount, shallow } from 'enzyme';
import ReactSixteenAdapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new ReactSixteenAdapter() })
import App from './app';
import itemData from './item-data';
import changedData from './changed-data'

describe('App', () => {
	it('should render App', () => {
    const data = itemData
		const app = mount(<App itemData={ data } />)
		expect(app.debug()).toMatchSnapshot();
	});
	it('render a title', () => {
		const app = shallow(<App itemData={ itemData } />);
		expect(app.find('.my-retail-title').text()).toEqual('Ninjaâ„¢ Professional Blender with Single Serve Blending Cups');
	});
	it('render an image', () => {
		const app = shallow(<App itemData={ itemData } />);
		expect(app.find('img').props().src).toEqual('http://target.scene7.com/is/image/Target/14263758');
	});
	it('should render image carousel', () => {
			const data = itemData
			const app = shallow(<App itemData={ data } />)
			expect(app.find('.my-retail-image-carousel').exists()).toBe(true);
		});
	it('render a price', () => {
		const app = shallow(<App itemData={ itemData } />);
		expect(app.find('.my-retail-offer-price').text()).toEqual('$139.99');
	});
	it('should render add to cart button when purchasingChannelCode equal to 0 or 1', () => {
			const data = itemData
			const app = shallow(<App itemData={ data } />)
			expect(app.find('.btn-add-to-cart').exists()).toBe(true);
		});
	it('should render pick up in store button when purchasingChannelCode equal to 0 or 2', () => {
			const data = itemData
			const app = shallow(<App itemData={ data } />)
			expect(app.find('.btn-pickup').exists()).toBe(true);
		});
		describe('Testing buttons with purchasingChannelCode not equal to 0 or 1', () => {
			it('should render all other elements except buttons', () => {
				const data = changedData
				const app = mount(<App itemData={ data } />)
				expect(app.debug()).toMatchSnapshot();
			});
			it('should not render pick up in store button', () => {
					const data = changedData
					const app = shallow(<App itemData={ data } />)
					expect(app.find('.btn-pickup').exists()).toBe(false);
				});
				it('should not render add to cart button', () => {
						const data = changedData
						const app = shallow(<App itemData={ data } />)
						expect(app.find('.btn-add-to-cart').exists()).toBe(false);
					});
			});
});
