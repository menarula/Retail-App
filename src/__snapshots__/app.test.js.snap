// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing buttons with purchasingChannelCode not equal to 0 or 1 should render all other elements except buttons 1`] = `
"<App itemData={{...}}>
  <div className=\\"row\\">
    <div className=\\"my-retail col-xs-12\\">
      <div className=\\"my-retail-product col-xs-12 col-md-6\\">
        <h1 className=\\"my-retail-title\\">
          Ninja™ Professional Blender with Single Serve Blending Cups
        </h1>
        <div className=\\"my-retail-product-image-wrapper\\">
          <img className=\\"my-retail-product-image\\" src=\\"http://target.scene7.com/is/image/Target/14263758\\" alt=\\"product\\" />
        </div>
        <div className=\\"my-retail-image-carousel\\">
          <Slider images={{...}}>
            <Carousel showArrows={true} showIndicators={false} showStatus={false} showThumbs={false} infiniteLoop={false} selectedItem={0} axis=\\"horizontal\\" verticalSwipe=\\"standard\\" width=\\"100%\\" useKeyboardArrows={false} autoPlay={false} stopOnHover={true} interval={3000} transitionTime={350} swipeScrollTolerance={5} dynamicHeight={false} emulateTouch={false} onClickItem={[Function]} onClickThumb={[Function]} onChange={[Function]} statusFormatter={[Function]} centerMode={false} centerSlidePercentage={80}>
              <div className={[undefined]}>
                <div className=\\"carousel carousel-slider\\" style={{...}}>
                  <button type=\\"button\\" className=\\"control-arrow control-prev control-disabled\\" onClick={[Function]} />
                  <div className=\\"slider-wrapper axis-horizontal\\" style={{...}}>
                    <ReactSwipe tagName=\\"ul\\" selectedItem={0} className=\\"slider animated\\" onSwipeMove={[Function]} onSwipeStart={[Function]} onSwipeEnd={[Function]} style={{...}} tolerance={5} onSwipeLeft={[Function]} onSwipeRight={[Function]} allowMouseEvents={false} onSwipeUp={[Function]} onSwipeDown={[Function]}>
                      <ul onMouseDown={[Function]} onTouchStart={[Function]} onTouchEnd={[Function]} className=\\"slider animated\\" style={{...}}>
                        <li className=\\"slide selected\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt01\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt02\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt03\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt04\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt05\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt06\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt07\\" />
                          </div>
                        </li>
                      </ul>
                    </ReactSwipe>
                  </div>
                  <button type=\\"button\\" className=\\"control-arrow control-next\\" onClick={[Function]} />
                </div>
              </div>
            </Carousel>
          </Slider>
        </div>
      </div>
      <div className=\\"my-retail-priceblock col-xs-12 col-md-6\\">
        <div className=\\"my-retail-price\\">
          <span className=\\"my-retail-offer-price\\">
            $139.99
          </span>
          <span className=\\"my-retail-price-qualifier\\">
            Online Price
          </span>
        </div>
        <div className=\\"my-retail-promotions col-xs-12 col-md-6\\">
          <p>
            spend $50, ship free
          </p>
          <p>
            $25 gift card with purchase of a select Ninja blender
          </p>
        </div>
        <div className=\\"my-retail-button-wrapper col-xs-12\\" />
      </div>
    </div>
  </div>
</App>"
`;

exports[`App should render App 1`] = `
"<App itemData={{...}}>
  <div className=\\"row\\">
    <div className=\\"my-retail col-xs-12\\">
      <div className=\\"my-retail-product col-xs-12 col-md-6\\">
        <h1 className=\\"my-retail-title\\">
          Ninja™ Professional Blender with Single Serve Blending Cups
        </h1>
        <div className=\\"my-retail-product-image-wrapper\\">
          <img className=\\"my-retail-product-image\\" src=\\"http://target.scene7.com/is/image/Target/14263758\\" alt=\\"product\\" />
        </div>
        <div className=\\"my-retail-image-carousel\\">
          <Slider images={{...}}>
            <Carousel showArrows={true} showIndicators={false} showStatus={false} showThumbs={false} infiniteLoop={false} selectedItem={0} axis=\\"horizontal\\" verticalSwipe=\\"standard\\" width=\\"100%\\" useKeyboardArrows={false} autoPlay={false} stopOnHover={true} interval={3000} transitionTime={350} swipeScrollTolerance={5} dynamicHeight={false} emulateTouch={false} onClickItem={[Function]} onClickThumb={[Function]} onChange={[Function]} statusFormatter={[Function]} centerMode={false} centerSlidePercentage={80}>
              <div className={[undefined]}>
                <div className=\\"carousel carousel-slider\\" style={{...}}>
                  <button type=\\"button\\" className=\\"control-arrow control-prev control-disabled\\" onClick={[Function]} />
                  <div className=\\"slider-wrapper axis-horizontal\\" style={{...}}>
                    <ReactSwipe tagName=\\"ul\\" selectedItem={0} className=\\"slider animated\\" onSwipeMove={[Function]} onSwipeStart={[Function]} onSwipeEnd={[Function]} style={{...}} tolerance={5} onSwipeLeft={[Function]} onSwipeRight={[Function]} allowMouseEvents={false} onSwipeUp={[Function]} onSwipeDown={[Function]}>
                      <ul onMouseDown={[Function]} onTouchStart={[Function]} onTouchEnd={[Function]} className=\\"slider animated\\" style={{...}}>
                        <li className=\\"slide selected\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt01\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt02\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt03\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt04\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt05\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt06\\" />
                          </div>
                        </li>
                        <li className=\\"slide\\" onClick={[Function]}>
                          <div>
                            <img src=\\"http://target.scene7.com/is/image/Target/14263758_Alt07\\" />
                          </div>
                        </li>
                      </ul>
                    </ReactSwipe>
                  </div>
                  <button type=\\"button\\" className=\\"control-arrow control-next\\" onClick={[Function]} />
                </div>
              </div>
            </Carousel>
          </Slider>
        </div>
      </div>
      <div className=\\"my-retail-priceblock col-xs-12 col-md-6\\">
        <div className=\\"my-retail-price\\">
          <span className=\\"my-retail-offer-price\\">
            $139.99
          </span>
          <span className=\\"my-retail-price-qualifier\\">
            Online Price
          </span>
        </div>
        <div className=\\"my-retail-promotions col-xs-12 col-md-6\\">
          <p>
            spend $50, ship free
          </p>
          <p>
            $25 gift card with purchase of a select Ninja blender
          </p>
        </div>
        <div className=\\"my-retail-button-wrapper col-xs-12\\">
          <button className=\\"btn btn-block btn-sm btn-primary btn-pickup\\">
            PICK UP IN STORE
          </button>
          <button className=\\"btn btn-block btn-sm btn-primary btn-add-to-cart\\">
            ADD TO CART
          </button>
        </div>
      </div>
    </div>
  </div>
</App>"
`;
